### PR TITLE
start v24.09.00 development [skip ci]

### PR DESCRIPTION
**Pull request by jameslamb**
#163 added support for `legate` / `cunumeric` 24.09 nightlies here.

This PR proposes the following:

* starting to version `legate-boost` packages as `24.09.00dev{n}`
* updating the instructions for releasing
* minor adjustments to how we pin to `legate` / `cunumeric` packages

## Notes for Reviewers

See inline comments for explanations of these changes.

### CODE REVIEW

File: conda/recipes/legate-boost/meta.yaml [hunk 1]
```diff
@@ -80,12 +79,12 @@ requirements:
     {% if gpu_enabled_bool %}
     - cuda-version ={{ cuda_version }}
     - cuda-cudart-dev
-    - cunumeric {{ legate_version }} =*_gpu
-    - legate {{ legate_version }} =*_gpu
+    - cunumeric {{ legate_version }} =*_gpu*
+    - legate {{ legate_version }} =*_gpu*
```
Comment (jameslamb) [inline conda/recipes/legate-boost/meta.yaml:83]: `*_gpu` = "ends with `_gpu`"

`*_gpu*` = "contains `_gpu` somewhere"

I think we want that slightly more flexible form, as there's no guarantee that the packages we install will always end with `_gpu` (even if the packages we currently *build* do).

Look at the diff from the PR that added these build string components to `legate` packages... there are some other things at the end of the build strings sometimes: https://github.com/nv-legate/legate.core.internal/pull/1164/files

File: conda/recipes/legate-boost/meta.yaml [hunk 2]
```diff
@@ -14,7 +14,6 @@
 #  '*_gpu' for systems with CUDA
 #
 # this mirrors the approach used by cunumeric / legate
-# (though as of 24.06, they do not append a '*_gpu' to the build string for GPU packages)
```
Comment (jameslamb) [inline conda/recipes/legate-boost/meta.yaml:17]: 🎉 https://github.com/nv-legate/legate.core.internal/pull/1164/files

**Review by seberg — APPROVED** [review]
LGTM, thanks.

**Comment by jameslamb** [comment]
Looks like the `[skip ci]` approach worked! No new build was triggered when this was merged to `main`: https://github.com/rapidsai/legate-boost/actions/workflows/build.yaml

Then I pushed a new tag to the repo, following these docs.

```shell
git checkout main
git pull upstream main
git tag -a v24.09.00dev -m "v24.09.00dev"
git push upstream v24.09.00dev
```

https://github.com/rapidsai/legate-boost/releases/tag/v24.09.00dev

That also did not trigger a new build, because of this regex that says "only do a new build on tags of releases"

https://github.com/rapidsai/legate-boost/blob/166c98d3c1d59219ca43aab72c162e5ac9beefa9/.github/workflows/build.yaml#L13-L15

I'd copied that regex from other RAPIDS repos ([e.g. this cudf example](https://github.com/rapidsai/cudf/blob/ef28cddeccbcc790e05dd49794ecdfcae4f008c2/.github/workflows/build.yaml#L7-L8)). There, it serves the purpose "don't run a new build when a non-release tag is pushed".

But I think here in this repo, since we'll now use a PR with `[skip ci]` in the commit message, we really do want to trigger the build job on any tag push. Otherwise, it'd take a PR like this one **and then a second PR** to get the first new packages produced. I'll push a follow-up with changes like that.

**Comment by jameslamb** [comment]
I just deleted that `v24.09.00dev` tag, in place of a `v24.09.00.dev` tag on the same commit.

```shell
git checkout main
git pull upstream main
git tag -d v24.09.00dev
git tag -a v24.09.00.dev -m "v24.09.00.dev"
git push upstream v24.09.00.dev
```

To match the way that `cunumeric` and `legate` are versions, where the version ends with a `.dev{n}`.

I'm updating the packaging stuff for this repo to reflect that, in #173.

**Pull request merged**